src/AbstractOperations/conditional_operations.jl

```diff
@@ -95,10 +95,10 @@ validate_condition(cond::Function, ::AbstractField) = cond
 
 function validate_condition(cond::AbstractArray, operand::AbstractField)
     if ndims(cond) != 3
-        throw(ArgumentError("The keyword argument condition::AbstractArray requires a 3D array of size $(size(operand))"))
+        throw(ArgumentError("The keyword argument condition::AbstractArray requires size $(size(operand))"))
     end
     if eltype(cond) !== Bool
-        throw(ArgumentError("The keyword argument condition::AbstractArray requires an array of booleans"))
+        throw(ArgumentError("The keyword argument condition::AbstractArray requires eltype(condition) == Bool"))
```

Comment:
This isn't really true because the actual requirement is that `ifelse` can be evaluated. But if we want to make this restriction we can. I'm a little worried it might mess things up, eg if you use `Reactant` you end up with `TracedRNumber{Bool}` (which also has an `ifelse` defined for it)

Reply:
When I try providing an array of ones and zeros I get:

```Julia
ERROR: LoadError: MethodError: no method matching ifelse(::Float64, ::Float64, ::Int64)
The function `ifelse` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  ifelse(::Bool, ::Any, ::Any)
   @ Base essentials.jl:796

Stacktrace:
  [1] getindex
    @ ~/Library/CloudStorage/OneDrive-TheUniversityofMelbourne/Documents/Research/Oceananigans.jl-v2/src/AbstractOperations/conditional_operations.jl:118 [inlined]
  [2] _getindex
    @ ./abstractarray.jl:1358 [inlined]
  [3] getindex
    @ ./abstractarray.jl:1312 [inlined]
  [4] macro expansion
    @ ./reducedim.jl:280 [inlined]
  [5] macro expansion
    @ ./simdloop.jl:77 [inlined]
...
```

Reply:
right but if you included another package like ```using GreatIfElseTools` then you can extend `ifelse`

So if you restrict inputs you want to be sure you really don't want anyone to use it any other way

Reply:
that's a general Julia concept that we don't want to over-restrict inputs. I don't know if there is an abstract boolean, that could be used...

Reply:
Agreed; I removed the bool validation.